src/Dapr.Actors/Client/ActorProxyFactory.cs

```diff
@@ -6,58 +6,56 @@
 namespace Dapr.Actors.Client
 {
     using System;
+    using System.Text.Json;
     using Dapr.Actors.Builder;
     using Dapr.Actors.Communication.Client;
 
     /// <summary>
     /// Represents a factory class to create a proxy to the remote actor objects.
     /// </summary>
-    internal class ActorProxyFactory : IActorProxyFactory
+    public class ActorProxyFactory : IActorProxyFactory
```

Comment:
What's the value in making this *class* public? vs the interface - do you have a code sample of how the class would be used?

I think what's missing here is that the `JsonSerializerOptions` should be a get/set property on the class. Then you can instantiate your own instance of `ActorProxyFactory` and tweak its settings and reuse that.

If that's the intended design then I think you'll want to:

- Turn `JsonSerializerOptions` into a get/set property
- Make `ActorProxyFactory` sealed
- Change the type of `ActorProxy.DefaultProxyFactory` to be `IActorProxyFactory` so that the default implementation isn't mutable.

Reply:
My thought was that we would use it in the `Startup.cs` or `Program.cs` via:
```C#
ActorProxy.DefaultProxyFactory.JsonSerializerOptions.Converters.Add(new JsonFSharpConverter())
```
I didn't change it to an interface because I wasn't sure on the original decision to have it be a class. I agree that it should be an interface so the user can just set it to which ever factory implementation they want. 

Changing it to an interface makes it so the example above won't work unless we add `JsonSerializerOptions` to the interface (which might be too prescriptive - ie. what if we want an XML Actor Factory?)

By changing it to an interface the above example becomes much more verbose:
```C#
var fSharpJsonSerializer = new JsonSerializerOptions {
                                   PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
                                   PropertyNameCaseInsensitive = true
                               };
fSharpJsonSerializer.Converters.Add(JsonFSharpConverter());

var proxyOptions = new ActorProxyOptions { JsonSerializerOptions = fSharpJsonSerializer };
var actorProxyFactory = ActorProxyFactory(proxyOptions);

ActorProxy.DefaultProxyFactory = actorProxyFactory;
```

In F#:
```F#
let fSharpJsonSerializer = JsonSerializerOptions()
fSharpJsonSerializer.PropertyNamingPolicy <- JsonNamingPolicy.CamelCase
fSharpJsonSerializer.PropertyNameCaseInsensitive <- true
fSharpJsonSerializer.Converters.Add(JsonFSharpConverter())

let proxyOptions = ActorProxyOptions()
proxyOptions.JsonSerializerOptions <- fSharpJsonSerializer
let actorProxyFactory = ActorProxyFactory(proxyOptions)

ActorProxy.DefaultProxyFactory <- actorProxyFactory
```

By making the `ActorProxy.JsonSerializerOptions` into a public property the user could simply set the serializer options on a per proxy basis via:
```C#
var proxy = ActorProxy.Create<IMyActor>(new ActorId(id), "MyActor");
proxy.JsonSerializerOptions.Converters.Add(JsonFSharpConverter());
```

Reply:
I hid my previous comment after thinking about it a little more it seems like it would be acceptable to add an `ActorProxyOptions` property to the `IActorProxyFactory` interface. This would make it so we set the default `JsonSerializerOptions` on new `ActorProxy` instances via the `DefaultActorProxyOptions` in the factory or via the `ActorProxy.Create` method. 

So in order to add a custom serializer to all `ActorProxy`'s by default you would do:
```C#
ActorProxy.DefaultProxyFactory.DefaultActorProxyOptions.JsonSerializerOptions.Converters.Add(new JsonFSharpConverter());
```

Reply:
> So in order to add a custom serializer to all ActorProxy's by default you

This is a case I want to avoid. No mutable statics allowed.

This has been addressed by removing the options property from `IActorProxyFactory`.